test/unit_test/static/quantity_kind_test.cpp

```diff
@@ -386,11 +386,18 @@ static_assert([]() {
   w = width<metre, int>(3 * m);
   assert(&(w *= 3.9) == &w && w.common() == 11 * m);
   assert(&(w /= 3.9) == &w && w.common() == 2 * m);
-  assert(&(w %= 3.9) == &w && w.common() == 2 * m);
   return true;
 }());
 #endif
 
+static_assert((std::uint8_t(255) * m %= 256) == (width<metre, std::uint8_t>(255 * m) %= 256).common());
+// static_assert((std::uint8_t(255) * m %= 256 * m) !=
+//   (width<metre, std::uint8_t>(255 * m) %= width<metre, std::uint8_t>(256 * m)).common());  // UB
+static_assert((std::uint8_t(255) * m %= 257) == (width<metre, std::uint8_t>(255 * m) %= 257).common());
+// TODO: Fix
```

Comment:
There's no TODO here. The test just ensures it does the same as `quantity`. It is `quantity` that needs to be fixed.